contrib/core-contract-tests/tests/pox-4/pox-4.prop.test.ts

```diff
@@ -0,0 +1,1839 @@
+            caller
+          );
+          // Assert
+          assert(isClarityType(actual, ClarityType.ResponseOk));
+          assert(isClarityType(actual.value, ClarityType.Tuple));
+          expect(actual.value.data["first-burnchain-block-height"]).toBeUint(
+            expected_first_burn_block_height
+          );
+          expect(actual.value.data["min-amount-ustx"]).toBeUint(
+            MIN_AMOUNT_USTX
+          );
+          expect(actual.value.data["prepare-cycle-length"]).toBeUint(
+            TESTNET_PREPARE_CYCLE_LENGTH
+          );
+          expect(actual.value.data["reward-cycle-id"]).toBeUint(
+            expected_reward_cycle_id
+          );
+          expect(actual.value.data["reward-cycle-length"]).toBeUint(
+            TESTNET_REWARD_CYCLE_LENGTH
+          );
+          expect(actual.value.data["total-liquid-supply-ustx"]).toBeUint(
+            INITIAL_TOTAL_LIQ_SUPPLY
+          );
+        }
+      )
+    );
+  });
+
+  it("should return none get-allowance-contract-caller", () => {
+    fc.assert(
+      fc.property(
+        fc.constantFrom(...simnet.getAccounts().values()),
+        fc.constantFrom(...simnet.getAccounts().values()),
+        fc.constantFrom(...simnet.getAccounts().values()),
+        (caller, sender, contract_caller) => {
+          // Arrange
+          // Act
+          const { result: actual } = simnet.callReadOnlyFn(
+            "pox-4",
+            "get-allowance-contract-callers",
+            [Cl.principal(sender), Cl.principal(contract_caller)],
+            caller
+          );
+          // Assert
+          assert(isClarityType(actual, ClarityType.OptionalNone));
+        }
+      )
+    );
+  });
+
+  it("should return some(until-burn-ht: none) get-allowance-contract-caller after allow-contract-caller", () => {
+    fc.assert(
+      fc.property(
+        fc.constantFrom(...simnet.getAccounts().values()),
+        fc.constantFrom(...simnet.getAccounts().values()),
+        fc.constantFrom(...simnet.getAccounts().values()),
+        (caller, sender, contract_caller) => {
+          // Arrange
+          const { result: allow } = simnet.callPublicFn(
+            "pox-4",
+            "allow-contract-caller",
+            [Cl.principal(contract_caller), Cl.none()],
+            sender
+          );
+          assert(isClarityType(allow, ClarityType.ResponseOk));
+          assert(isClarityType(allow.value, ClarityType.BoolTrue));
+          // Act
+          const { result: actual } = simnet.callReadOnlyFn(
+            "pox-4",
+            "get-allowance-contract-callers",
+            [Cl.principal(sender), Cl.principal(contract_caller)],
+            caller
+          );
+          // Assert
+          assert(isClarityType(actual, ClarityType.OptionalSome));
+          assert(isClarityType(actual.value, ClarityType.Tuple));
+          expect(actual).toBeSome(Cl.tuple({ "until-burn-ht": Cl.none() }));
+          expect(actual.value).toBeTuple({ "until-burn-ht": Cl.none() });
```

Comment:
This line causes the issue reported by @setzeus above. However, this line can be safely removed since the exact same assertion happens also in the line above.

(What's the reason for having the same assertion twice?)
```suggestion

```

@hugocaillard, this also resolves the question in https://github.com/stacks-network/stacks-core/pull/4470/files#r1519481753 so feel free to skip/ignore it.